tests/test_lint.py

```diff
@@ -287,3 +287,248 @@ def test_catalog_name():
+        "title": "Simple Example Collection",
+        "providers": [
+            {
+            "name": "Remote Data, Inc",
+            "description": "Producers of awesome spatiotemporal assets",
+            "roles": [
+                "producer",
+                "processor"
+            ],
+            "url": "http://remotedata.io"
+            }
+        ],
+        "extent": {
+            "spatial": {
+            "bbox": [
+                [
+                172.91173669923782,
+                1.3438851951615003,
+                172.95469614953714,
+                1.3690476620161975
+                ]
+            ]
+            },
+            "temporal": {
+            "interval": [
+                [
+                "2020-12-11T22:38:32.125Z",
+                "2020-12-14T18:02:31.437Z"
+                ]
+            ]
+            }
+        },
+        "license": "CC-BY-4.0",
+        "summaries": {
+            "platform": [
+            "cool_sat1",
+            "cool_sat2"
+            ],
+            "constellation": [
+            "ion"
+            ],
+            "instruments": [
+            "cool_sensor_v1",
+            "cool_sensor_v2"
+            ],
+            "gsd": {
+            "minimum": 0.512,
+            "maximum": 0.66
+            },
+            "eo:cloud_cover": {
+            "minimum": 1.2,
+            "maximum": 1.2
+            },
+            "proj:epsg": {
+            "minimum": 32659,
+            "maximum": 32659
+            },
+            "view:sun_elevation": {
+            "minimum": 54.9,
+            "maximum": 54.9
+            },
+            "view:off_nadir": {
+            "minimum": 3.8,
+            "maximum": 3.8
+            },
+            "view:sun_azimuth": {
+            "minimum": 135.7,
+            "maximum": 135.7
+            }
+        },
+        "links": [
+            {
+            "rel": "root",
+            "href": "./collection.json",
+            "type": "application/json",
+            "title": "Simple Example Collection"
+            },
+            {
+            "rel": "item",
+            "href": "./simple-item.json",
+            "type": "application/geo+json",
+            "title": "Simple Item"
+            },
+            {
+            "rel": "item",
+            "href": "./core-item.json",
+            "type": "application/geo+json"
+            },
+            {
+            "rel": "item",
+            "href": "./extended-item.json",
+            "type": "application/geo+json",
+            "title": "Extended Item"
+            }
+        ]
+    }
+    linter = Linter(file)
+    assert linter.valid_stac == True
+    assert linter.asset_type == "COLLECTION"
+    assert linter.check_catalog_id_file_name() == False
+    assert linter.create_best_practices_dict()["check_catalog_id"] == ["Object should be called 'collection.json' not 'simple-collection.json'"]
```

Comment:
Collections usually have descriptive IDs, e.g. https://planetarycomputer.microsoft.com/api/stac/v1/collections/sentinel-2-l2a. I think this lint is incorrect.

Reply:
This project was an attempt to follow the STAC Best Practices document. I think in the document this advice is only for static catalogs? https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#static-catalogs

Reply:
Here too: https://github.com/radiantearth/stac-spec/blob/master/best-practices.md#catalog-layout

Reply:
Thanks for this. I'm changing the check so that it only activates with a static file.